src/python/gudhi/persistence_graphical_tools.py

```diff
@@ -321,11 +326,16 @@ def plot_persistence_diagram(
         # line display of equation : birth = death
         axes.plot([axis_start, axis_end], [axis_start, axis_end], linewidth=1.0, color="k")
 
-        x=[birth for (dim,(birth,death)) in persistence]
-        y=[death if death != float("inf") else infinity for (dim,(birth,death)) in persistence]
-        c=[colormap[dim] for (dim,(birth,death)) in persistence]
+        non_empty_diagram = len(persistence[0]) > 0
```

Comment:
Why does it look at `len(persistence[0])`? I could understand `len(persistence)`, although I would be a bit surprised.

Reply:
You are completely right. I modified code and test on [393216c](https://github.com/GUDHI/gudhi-devel/pull/604/commits/393216cc99af79526b1bb90ae4da21116884bcbd)